docs/developer-docs/latest/update-migration-guides/migration-guides/v4/code/dependencies.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Code migration - Dependencies - Strapi Developer Docs
+description: Migrate your dependencies from Strapi v3.6.8 to v4.0.x with step-by-step instructions
+canonicalUrl:  Used by Google to index page, should start with https://docs.strapi.io/ — delete this comment when done [paste final URL here]
+---
+
+<!-- TODO: update SEO -->
+
+# v4 code migration: Updating dependencies
+
+!!!include(developer-docs/latest/update-migration-guides/migration-guides/v4/snippets/code-migration-intro.md)!!!
+
+:::strapi v3/v4 comparison
+In Strapi v3, Strapi package names were prefixed with `strapi-`.
+
+Strapi v4 uses scoped packages.
+:::
+
+To migrate dependencies to Strapi v4, update all references to Strapi packages using the new names found in the following table, and make sure the version number in `package.json` is up-to-date.
```

Comment:
This isn't entirely true, you don't need to add all of the same packages anymore, most you don't need to strictly define anymore. Only the `@strapi/strapi` is required now.

By default we generate new apps with:

- `@strapi/strapi`
- `@strapi/plugin-users-permissions`
- `@strapi/plugin-i18n`

---

The following don't need to be manually added anymore:

- `@strapi/database`
- `@strapi/admin`
- `@strapi/utils`
- `@strapi/helper-plugin`
- `@strapi/plugin-upload`
- `@strapi/plugin-email`
- `@strapi/plugin-content-type-builder`
- `@strapi/plugin-content-manager`

And if anything I wouldn't recommend people add these to their package.json as it would cause conflicts with how we load them now. (unless they are forking them which goes way beyond this guide)

Reply:
Here is a sample package.json from a fresh project:

```json
{
  "name": "test-lifecycles",
  "private": true,
  "version": "0.1.0",
  "description": "A Strapi application",
  "scripts": {
    "develop": "strapi develop",
    "start": "strapi start",
    "build": "strapi build",
    "strapi": "strapi"
  },
  "devDependencies": {},
  "dependencies": {
    "@strapi/strapi": "4.0.7",
    "@strapi/plugin-users-permissions": "4.0.7",
    "@strapi/plugin-i18n": "4.0.7",
    "sqlite3": "5.0.2"
  },
  "author": {
    "name": "A Strapi developer"
  },
  "strapi": {
    "uuid": "f9fa59de-0b06-4b53-b9b2-c3358a0baf56"
  },
  "engines": {
    "node": ">=12.x.x <=16.x.x",
    "npm": ">=6.0.0"
  },
  "license": "MIT"
}
```

Reply:
Thank you for the explanation, Derrick.

If someone has any package featured in the table already listed in their dependencies, they still need to update the name, right?

If not, what would you recommend people do to update their dependencies?

Reply:
They should remove it if it's not required, not just update the version.

upload, email, ctb, ctm, helper, utils, admin, database all should be removed unless the user forked them (which in the scope of this guide, they haven't)